packages/s3-credentials-endpoint/index.js

```diff
@@ -92,6 +95,112 @@ async function displayS3CredentialInstructions(_req, res) {
   res.send(compiled(process.env));
 }
 
+/**
+ * If DISABLE_S3_CREDENTIALS is not "true", returns undefined, otherwise, send a
+ * boom.ServerUnavailable to the caller, Exiting the request.
+ *
+ * @param {Object} res - express request object
+ * @returns {undefined} - when DISABLE_S3_CREDENTIALS is not 'true'
+ */
+function ensureEndpointEnabled(res) {
+  const disableS3Credentials = process.env.DISABLE_S3_CREDENTIALS;
+
+  if (disableS3Credentials && (disableS3Credentials.toLowerCase() === 'true')) {
+    return res.boom.serverUnavailable('S3 Credentials Endpoint has been disabled');
+  }
+  return undefined;
+}
+
+/**
+ * @returns {bool} whether or not the endpoint is configured to send ACL based
+ * credentials.
+ */
+function configuredForACLCredentials() {
+  if (process.env.CMR_ACL_BASED_CREDENTIALS && process.env.CMR_ACL_BASED_CREDENTIALS.toLowerCase() === 'true') {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Parses a "bucket/key/path" to return an array of ["bucket", "key/path"] where
+ * keypath is "/" if not specified.
+ *
+ * @param {string} bucketKeyPath
+ * @returns {Array<string>} Array [bucket, keypath]
+ */
+function parseBucketKey(bucketKeyPath) {
+  try {
+    const parts = bucketKeyPath.split('/');
+    const bucket = parts.shift();
+    const keypath = parts.join('/');
+    return [bucket, `/${keypath}`];
+  } catch (error) {
+    return [undefined, undefined];
+  }
+}
+
+/**
+ * Reformats a list of buckets and bucket/keyspaths into the shape needed for
+ * calling the sts policy helper function. The desired payload is an object
+ * with 3 keys, 'accessmode', 'bucketlist' and 'pathlist'. 'bucketlist' and
+ * 'pathlist' are arrays of matching bucket and paths.
+ *
+ * Example:
+ * if the cmrAllowedBucketKeyList is:
+ * [ 'bucket1/somepath', 'bucket2']
+ * then the desired output would be the strigified object:
+ *
+ *  {
+ *   accessmode: 'Allow',
+ *   bucketlist: ['bucket1','bucket2'],
+ *   pathlist: ['/somepath', '/']
+ *  }
+ *
```

Comment:
For the cmr bucket key list ["cumulus-sandbox-public/*","cumulus-test-sandbox-protected-2/*"], how do you handle the `*` key?

Reply:
I treat the * as any other path, I can add one to the test no problem.

Reply:
Maybe this isn't what you meant? 5e841bea26

Reply:
I know we have tested the bucket access.  Have you tested with the ACL bucket/path, like mhs3-protected/allowedpath, and make sure the user only has access to the allowedpath?